src/OpenTelemetry.AutoInstrumentation/ContinuousProfiler/ContinuousProfilerProcessor.cs

```diff
@@ -86,13 +94,20 @@ private static void ActivityChanged(AsyncLocalValueChangedArgs<Activity?> sender
         NativeMethods.ContinuousProfilerSetNativeContext(0, 0, 0);
     }
 
-    private static void SampleReadingThread(BufferProcessor sampleExporter, TimeSpan exportInterval)
+    private void SampleReadingThread(BufferProcessor sampleExporter, TimeSpan exportInterval)
     {
         while (true)
         {
-            // TODO Graceful shutdown and Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
-            Thread.Sleep(exportInterval);
+            // TODO Task.Delay https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/issues/3216
+            var shutdownRequested = _shutdownTrigger.Wait(exportInterval);
+
             sampleExporter.Process();
```

Comment:
This call may never exit. `sampleExporter.Process()` can starve the shutdown as there is no timeout nor cancelation handling.

Ideas how this can be handled:
1. We can pass a "export timeout" or CancelationToken to the `threadSamplesMethod` and `allocationSamplesMethod` methods. The responsibility of cancelling is moved to the called method (plugin). The good think is that we will get e.g. logs what happened wrong.
2. We can also `_thread?.Abort()` the thread. I think this is a "last resort" if the plugin does handles the cancelation properly. We can execute it e.g. if the plugin does not cancel in `2 * exportTimeout`.

Reply:
That's a very good point.
Option 2 won't work as `Thread.Abort` is [not supported](https://learn.microsoft.com/en-us/dotnet/api/system.threading.thread.abort?view=net-8.0) for .NET5+.
That leaves us with option 1, will update the PR accordingly.

Reply:
We discussed that instead of option 2 we can simply replace `_thread?.Join()` with `_thread?.Join(exportTimeout)`

Reply:
We can start with option 2 and apply option 1 in a separate PR.

Reply:
fixed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3294/commits/0b6fd27fbeee942c36031a6987919cc79b918eb8